Adding possibility to group speakers from different platform than sonos

hcoohb opened the pull request:
Hi,

I love your mini-media-player card and its functionalities!
So much that I found it is now worth implementing the grouping of speakers for the yamaha_musiccast platform, now that we have a nice and intuitive interface.

As the speaker groups are currently only targetting sonos, could we have a more generic approach by specifying the platform?

I have developed the grouping services in the yamaha_musiccast exactly the same as the sonos join/unjoin services.
My implementation is here: [Update for yamaha_musiccast platform](https://github.com/hcoohb/home-assistant/tree/update-musiccast) 
I have not yet pushed it to hass as PR, as my PR on the pymusiccast module needs to happen first [here](https://github.com/jalmeroth/pymusiccast/pull/13)

I have tried to extend your work to have other types of speakers allowed to use the grouping function.
It works as it is with my yamaha speakers.
Let me know what you think.

Thanks and regards,
Fabien

kalkih commented:
Very cool. nice work on all of this!

Does the music cast group system work similar to Sonos in the case of the coordinator/master speaker? In Sonos the master speaker is always listed as the first entity in the `sonos_group` attribute.

I'm trying to stay away from breaking changes to config options, or at least deprecate them before removing them completely. Adding a small guard in setConfig for `sonos` and make `sonos` the default platform would make your changes non-breaking.

Could you also update the options in the readme to reflect the changes in this PR?

kalkih commented on src/main.js:
```diff
@@ -109,9 +109,9 @@ class MiniMediaPlayer extends LitElement {
+      speaker_group: {
         show_group_count: true,
-        ...config.sonos,
+        ...config.speaker_group,
```
```js
...config.sonos,
...config.speaker_group,
```
to avoid breaking change, what do you think?

hcoohb replied:
Makes perfect sense. added in new commit

kalkih commented on src/main.js:
```diff
@@ -109,9 +109,9 @@ class MiniMediaPlayer extends LitElement {
       hide: { ...DEFAULT_HIDE, ...config.hide },
-      sonos: {
+      speaker_group: {
         show_group_count: true,
```
```js
platform: 'sonos',
```
default to sonos to avoid breaking change

hcoohb replied:
Added in new commit

hcoohb commented:
Yes, for the yamaha_musiccast, I have also implemented such as the master is always the first one listed. 
Backward compatibility the way you mentioned is really nice! Added in a commit.
I have updated the readme as well but the `yamaha_musiccast` componenet from HASS is not working yet as the services have not been merged yet.

Thanks

kalkih commented:
Awesome, great job!

Thanks for the PR.

jodur commented:
Did anything changes concering the call/naming convention  trough the join/unjoin services for yamaha music_cast?
I used @hcoohb  modified version for yamaha musiccast and it worked without problems.
After updating to the latest version with HACS, the service calls shows an error.
I still can call the services trough the home-assistant developer tools.

example:

media_player.yamaha_musiccast_join
Service data:
entity_id: media_player.keuken_main
master: media_player.woonkamer_main
